.github/workflows/task-test.yml

```diff
@@ -80,26 +75,59 @@ jobs:
             tests/pytests/requirements.*
       - name: Deps checkout (node20 not supported)
         if: steps.node20.outputs.supported == 'false'
-        uses: actions/checkout@v3
-        with:
-          path: setup
-          sparse-checkout-cone-mode: false
-          sparse-checkout: |
-            .install
-            tests/pytests/requirements.*
-      - name: Setup specific
+        run: |
+          # Execute the logic based on the detected platform
+          echo "Detected platform: ${{ inputs.container }}"
```

Comment:
Can all of these bash scripts move to separate yml files?
It might also allow some code reuse.

Reply:
This is before we have our repo cloned, so I don't think so. At this point the machine only has this flow in hand.

Reply:
Maybe we can hold scripts in the repo and use them like variables

Reply:
How does this script get called?
Isn't it used like this?
```
uses: ./.github/workflows/task-test.yml
```

Reply:
Now I see your meaning - sounds good will try to add it.